src/hooks/useApi.tsx

```diff
@@ -0,0 +1,67 @@
+import type { Api } from '@jellyfin/sdk';
+import type { UserDto } from '@jellyfin/sdk/lib/generated-client';
+import type { ApiClient, Event } from 'jellyfin-apiclient';
+import React, { createContext, FC, useContext, useEffect, useState } from 'react';
+
+import ServerConnections from '../components/ServerConnections';
+import events from '../utils/events';
+import { toApi } from '../utils/jellyfin-apiclient/compat';
+
+interface JellyfinApiContext {
+    __legacyApiClient__?: ApiClient
+    api?: Api
+    user?: UserDto
+}
+
+export const ApiContext = createContext<JellyfinApiContext>({});
+export const useApi = () => useContext(ApiContext);
+
+export const ApiProvider: FC = ({ children }) => {
+    const [ legacyApiClient, setLegacyApiClient ] = useState<ApiClient>();
+    const [ api, setApi ] = useState<Api>();
+    const [ user, setUser ] = useState<UserDto>();
+
+    useEffect(() => {
+        ServerConnections.currentApiClient()
```

Comment:
```suggestion
const apiClient = ServerConnections.currentApiClient();
        if (apiClient) {
            apiClient.getCurrentUser()
                .then(newUser => updateApiUser(undefined, newUser))
                .catch(err => {
                    console.info('[ApiProvider] Could not get current user', err);
                });
        }```

Reply:
@thornbill this solve the problem

Reply:
Should be fixed now. :crossed_fingers: I updated the return types for `currentApiClient()` to include `undefined` and fixed the related issues.